specification/types.md

```diff
@@ -122,8 +126,19 @@ A structure containing the following fields:
 
 A structure which supports definition of arbitrary properties, with keys of type `string`, and values of type `boolean`, `string`, or `number`.
 
+`key` is a required property with a value of `string`.
+`type` is a required property with a `string` value of either `boolean`, `string`, `number`, or `structure`.
```

Comment:
Perhaps this should be an optional property. It's nice to have but may not be easily available to all providers.

Reply:
I'm okay with it being optional 👍🏻